DEV/org.openl.core/src/org/openl/binding/impl/MethodNodeBinder.java

```diff
@@ -121,6 +124,55 @@ public IBoundNode bind(ISyntaxNode node, IBindingContext bindingContext) throws
         }
     }
 
+    private void validateMethodParameter(IMethodCaller methodCaller, IBoundNode[] children, ISyntaxNode node, IBindingContext bindingContext) {
+        IOpenClass[] parameterTypes = methodCaller.getMethod().getSignature().getParameterTypes();
+        if (parameterTypes.length == children.length) {
+            for (int i = 0; i < parameterTypes.length; i++) {
+                if (parameterTypes[i].getDomain() != null) {
+                    IDomain<Object> domain = (IDomain<Object>) parameterTypes[i].getDomain();
+                    String invalidValuesString = null;
+                    if (children[i] instanceof LiteralBoundNode) {
+                        LiteralBoundNode literalBoundNode = (LiteralBoundNode) children[i];
+                        if (literalBoundNode.getValue() != null && !domain.selectObject(literalBoundNode.getValue())) {
+                            invalidValuesString = literalBoundNode.getValue().toString();
+                        }
+                    } else if (children[i] instanceof ArrayInitializerNode) {
+                        // In case of MultiCallOpenMethod
+                        StringBuilder invalidValues = new StringBuilder();
+                        validateParameterArray(((ArrayInitializerNode) children[i]).children, domain, invalidValues);
+                        if (invalidValues.length() > 0) {
+                            invalidValuesString = invalidValues.substring(0, invalidValues.length() - 1);
+                        }
+                    }
+                    if (invalidValuesString != null) {
+                        BindHelper.processError(String.format(
+                                "Object '%s' is outside of valid domain '%s'. The comparison always returns %s.",
+                                invalidValuesString,
+                                DomainUtils.toString(domain),
+                                "ne".equals(methodCaller.getMethod().getName())), node, bindingContext);
+                    }
+                }
+            }
+        }
+
+    }
+
+    private void validateParameterArray(IBoundNode[] iBoundNode, IDomain<Object> domain, StringBuilder invalidValues) {
+        if (iBoundNode.length > 0 && iBoundNode[0] instanceof LiteralBoundNode) {
+            for (IBoundNode boundNode : iBoundNode) {
+                LiteralBoundNode literalBoundNode = (LiteralBoundNode) boundNode;
+                if (literalBoundNode.getValue() != null && !domain.selectObject(literalBoundNode.getValue())) {
+                    invalidValues.append(literalBoundNode.getValue()).append(",");
```

Comment:
Each array element must be logged separatelly, without concatenation via comma.
Instead of the
`Object '125,225' is outside of valid domain '[100, 200, 300]'`
It should be
`Object '125' is outside of valid domain '[100, 200, 300]'`
`Object '225' is outside of valid domain '[100, 200, 300]'`